andela/mazus-ah-backend
# #167165065 Implement Article Comments

### What does this PR do?
- Enable comments on articles

#### Description of Task to be completed?
- Redefine associations between comments, users and articles
- Add /articles/:slug/comments endpoint for posting comments
- Validate user input
- Check db for article
- Create new entry in comments table for the given article
- Return newly created comment
- Modify any fetchArticles controller to include comments

#### How should this be manually tested?
- Log in and send a post request to api/v1/articles/:slug/comments 
  > - Note: replace `slug` with an existing article slug in the db
  > - `req.body` should have a property called `body` as the body of the comment
- When done, fetch the article and see the comment appended to the `articlecomments` array
#### Any background context you want to provide?
-  N/A
#### What are the relevant pivotal tracker stories?
- [#167165065](#167165065)

#### Screenshots (if appropriate)
<img width="737" alt="Screenshot 2019-08-02 at 3 37 47 PM" src="https://user-images.githubusercontent.com/20531075/62377868-845e9600-b53b-11e9-9189-63ebb1f13e59.png">
